client/components/TwoFactorModal/TwoFactorEmail.tsx

```diff
@@ -0,0 +1,76 @@
+import { Box, TextInput, Icon } from '@rocket.chat/fuselage';
+import { useAutoFocus } from '@rocket.chat/fuselage-hooks';
+import React, { ReactElement, useState, ChangeEvent } from 'react';
+
+import { useEndpoint } from '../../contexts/ServerContext';
+import { useToastMessageDispatch } from '../../contexts/ToastMessagesContext';
+import { useTranslation } from '../../contexts/TranslationContext';
+import GenericModal from '../GenericModal';
+import { Method, OnConfirm } from './TwoFactorModal';
+
+type TwoFactorEmailProps = {
+	onConfirm: OnConfirm;
+	onClose: () => void;
+	emailOrUsername: string;
+};
+
+const TwoFactorEmail = ({
```

Comment:
I think the component name should be more descriptive like "TwoFactorEmailModal"